modules/packages/LinearAlgebra.chpl

```diff
@@ -2064,6 +2326,114 @@ proc kron(A: [?ADom] ?eltType, B: [?BDom] eltType) {
 }
 
 
+/*
+  Matrix exponential using Pade approximation
+
+  Params:
+  A: Expects an N*N square matrix
+  use_exact_onenorm: boolean value specifying if the onenorm has to be exact.
+  Defaults to true.
+
+  TODO: Need to write a method to find approximate norm.
+  Currently only supports for use_exact_onenorm = true.
+
+  Returns:
+  expA: N*N matrix which is Matrix exponential of `A`
+
+  References:
+  [1] Awad H. Al-Mohy and Nicholas J. Higham (2009)
+  "A New Scaling and Squaring Algorithm for the Matrix Exponential."
+  SIAM Journal on Matrix Analysis and Applications.
+  31 (3). pp. 970-989. ISSN 1095-7162
+*/
+
+proc expm(A: [], param use_exact_onenorm=true){
+
+  if A.rank != 2 || A.domain.dim(0) != A.domain.dim(1) then {
+    writeln('Square Matrix Expected');
```

Comment:
It seems like this would be an appropriate time to `throw` an error, see [the error handling primer](https://chapel-lang.org/docs/latest/primers/errorHandling.html)